fiat-roles/src/main/java/com/netflix/spinnaker/fiat/providers/BaseProvider.java

```diff
@@ -38,6 +41,35 @@
 
   private Cache<Integer, Set<R>> cache = buildCache(20);
 
+  private final List<ResourceInterceptor> resourceInterceptors;
+
+  public BaseProvider() {
+    this(Collections.emptyList());
+  }
+
+  public BaseProvider(List<ResourceInterceptor> resourceInterceptors) {
+    Class<R> resourceType = resolveResourceType(getClass());
+    this.resourceInterceptors =
+        resourceInterceptors == null
```

Comment:
IMO, accepting `null` (especially in component constructors) just seems like encouraging bad behavior. People shouldn't be handing you a null object, and I don't think there's any reason that would happen as the code is written currently, so just don't allow it.

Reply:
Fair enough - first pass through this I tried to keep everything non breaking (hence the extra constructors etc) as I was working on it, but I can definitely unwind that